Signal/src/ViewControllers/ProfileViewController.m

```diff
@@ -29,28 +44,47 @@ @interface ProfileViewController () <UITextFieldDelegate, AvatarViewHelperDelega
 
 @property (nonatomic) BOOL hasUnsavedChanges;
 
+@property (nonatomic) ProfileViewMode profileViewMode;
+
 @end
 
 #pragma mark -
 
 @implementation ProfileViewController
 
+- (instancetype)initWithMode:(ProfileViewMode)profileViewMode
+{
+    self = [super init];
+
+    if (!self) {
+        return self;
+    }
+
+    self.profileViewMode = profileViewMode;
+
+    // Use the TSStorageManager.dbReadWriteConnection for consistency with the reads below.
+    [[[TSStorageManager sharedManager] dbReadWriteConnection] setDate:[NSDate new]
```

Comment:
In my mind using the globally shared dbConnection's make things more complicated to reason about WRT to unexpected blocking (so they should be avoided), and it's simpler to have a dedicated connection for independent actors.

Can you tell me how you're thinking about it - e.g. When is it preferable to use the global dbConnections and when is it preferable to build a dedicated dbConnection?